src/app/settings/views/Configuration/Security/Security.tsx

```diff
@@ -5,29 +5,46 @@ import { useDispatch, useSelector } from "react-redux";
 
 import TLSDisabled from "./TLSDisabled";
 import TLSEnabled from "./TLSEnabled";
+import VaultSettings from "./VaultSettings";
 
 import { useWindowTitle } from "app/base/hooks";
 import { actions as generalActions } from "app/store/general";
-import { tlsCertificate as tlsCertificateSelectors } from "app/store/general/selectors";
+import {
+  tlsCertificate as tlsCertificateSelectors,
+  vaultEnabled as vaultEnabledSelectors,
+} from "app/store/general/selectors";
 
 const Security = (): JSX.Element => {
   const dispatch = useDispatch();
   const tlsCertificate = useSelector(tlsCertificateSelectors.get);
   const tlsCertificateLoaded = useSelector(tlsCertificateSelectors.loaded);
+  const vaultEnabledLoaded = useSelector(vaultEnabledSelectors.loaded);
   useWindowTitle("Security");
 
   useEffect(() => {
     dispatch(generalActions.fetchTlsCertificate());
+    dispatch(generalActions.fetchVaultEnabled());
   }, [dispatch]);
 
-  if (!tlsCertificateLoaded) {
+  if (!tlsCertificateLoaded || !vaultEnabledLoaded) {
```

Comment:
It might be better to display separate loading indicators for each so that they can be loaded incrementally (correct me if I'm wrong, but I don't think they come through a single request/websocket message).

Reply:
Fixed this one. The whole page used to just depend on TLS being loaded since it was the only content, but I've added a separate spinner for TLS so that Secret Storage and Security Protocols can be displayed independently, regardless of whether the other has loaded.